### PR TITLE
feat: add more protobuf types part 2

### PR DESCRIPTION
**Pull request by djordon**
## Description

This PR pushes https://github.com/stacks-network/sbtc/issues/412 further along and a follow-up to https://github.com/stacks-network/sbtc/pull/558.

The documentation was taken from the existing structs.

## Changes

* Add over a some transaction message types.
* Add a recoverable signature message type.
* Split things up into different files.

## Testing Information

There protobufs aren't used, so there are no tests at the moment.

## Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: protobufs/stacks/common.proto [hunk 1]
```diff
@@ -2,23 +2,45 @@ syntax = "proto3";
...
+
+// The id for a transaction on the bitcoin blockchain.
+message BitcoinTxid {
+  crypto.Uint256 txid = 1;
+}
+
+// A bitcoin block hash.
+message BitcoinBlockHash {
+  crypto.Uint256 block_hash = 1;
+}
```
Comment (cylewitruk) [inline protobufs/stacks/common.proto:15]: Feels a _little_ weird that these are part of the `stacks` package

**Review by cylewitruk — COMMENTED** [review]
Looks good in general, since these aren't used (yet) I'm guessing that these will be iterated upon later. Just had one comment re: the bitcoin types.

File: protobufs/stacks/common.proto (hunk 1 above)
Comment (djordon) [reply protobufs/stacks/common.proto:15]: Yeah I was thinking the same thing. When I was opening the PR. Figured I'd move them later. I'll just move them now.

Comment (djordon) [reply protobufs/stacks/common.proto]: Fixed in https://github.com/stacks-network/sbtc/pull/577/commits/291b72e10d107d675ff7b553618a73410c45cb13

**Review by cylewitruk — APPROVED** [review]
looks good :+1:

**Pull request merged**